tests/unit/test_vhdl_parser.py

```diff
@@ -298,7 +298,8 @@ def test_parsing_entity_with_ports(self):
             """\
 entity name is
    port (clk : in std_logic;
-         data : out std_logic_vector(11-1 downto 0));
+         data : out std_logic_vector(11-1 downto 0);
+         data2 : in signal_type);
```

Comment:
You should also add a port which starts with `signal name : ...`. Not often that you that but it's legal. Then you should add one port that has a tab character or another whitespace between `signal` and `name` :smile:

Reply:
Good points - but before I try to catch the rest of the world here: What is the intention of removing the "signal" string in this function? I think this is parsed as a port declaration, so no signal keyword here anyway, right?

Reply:
Added a regex to catch this quite generic.